client/block.c

```diff
@@ -803,21 +793,15 @@ static int add_block_nolock(struct xdag_block *newBlock, xdag_time_t limit)
 				tmpNodeBlock.flags & BI_EXTRA ? ORPHAN_REMOVE_EXTRA : ORPHAN_REMOVE_NORMAL);
 
 		if(tmpNodeBlock.linkamount[i]) {
-			blockRef = tmpNodeBlock.link[i];
-			if(!blockRef->backrefs || blockRef->backrefs->backrefs[N_BACKREFS - 1]) {
-				struct block_backrefs *back = xdag_malloc(sizeof(struct block_backrefs));
-				if(!back) {
-					xdag_err("xdag_malloc failed. [add_block_nolock:%d]", __LINE__);
-					continue;
-				}
-				memset(back, 0, sizeof(struct block_backrefs));
-				back->next = blockRef->backrefs;
-				blockRef->backrefs = back;
-			}
-
-			for(j = 0; blockRef->backrefs->backrefs[j]; ++j);
-
-			blockRef->backrefs->backrefs[j] = nodeBlock;
+		        blockRef = tmpNodeBlock.link[i];
+			add_backref(blockRef, nodeBlock);
+		}
+	}
+	
+	if((nodeBlock->flags & BI_REMARK) & ~BI_EXTRA){
```

Comment:
remark can be added if not BI_EXTRA because it is sure that it will be added to the dag
BI_EXTRA block will be added later, *only* if it will become not BI_EXTRA